Features/Context/PlatformUI.php

```diff
@@ -59,28 +61,31 @@ class PlatformUI extends Context
     private $newPathsMap = array();
 
     /**
-     * Waits for Javascript to finish by checking the loading tags of the page.
+     * Behat spin functions
+     * causes waiting while a a certain function does not return true
+     * waits while an element is not present.
      */
-    protected function waitForLoadings()
+    public function spin($lambda, $wait = 30)
     {
-        $page = $this->getSession()->getPage();
-        $loadingClasses = array(
-            '.yui3-app-transitioning',
-            '.is-app-loading',
-            '.is-app-transitioning',
-            // content tree
-            '.ez-view-treeactionview.is-expanded .ez-view-treeview:not(.is-tree-loaded)',
-            '.ez-view-universaldiscoverybrowseview .ez-view-treeview:not(.is-tree-loaded)',
-            '.is-tree-node-loading',
-            // contenttype menu
-            '.ez-view-createcontentactionview.is-expanded:not(.is-contenttypeselector-loaded)',
-        );
-        $loadingSelector = implode(',', $loadingClasses);
-        while ($page->find('css', $loadingSelector) != null) {
-            usleep(100 * 1000); // 100ms
+        for ($i = 0; $i < $wait; ++$i) {
```

Comment:
This might be useful, it most likely does the exact same thing, just with a javascript check instead of xpath via mink. So might be faster.

@miguelcleverti For this to work, assuming it is still in Behat 3.0 / Mink 1.6. As we don't have jQuery, and should not hard code use of framework anyway, you'll need to use the DOM. Modern DOM supported by all browsers we support is rather straight forward using: getElementById, getElementsByTagName, getElementsByClassName, or querySelectorAll *(for more advance expressions, it can do anything, but the once above is more efficient if they fit)*.